### PR TITLE
Error 500 in Module Manager

### PR DESCRIPTION
**Pull request by flancer64**
### Description
There is an error in Module Manager (http://store.com/setup/index.php/moduleGrid/modules) if some module has no "moduleName" property:
```
{
  "name": "amzn\/amazon-pay-and-login-magento-2-module",
  "type": "magento2-module",
  "version": "2.0.4"
}
```

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/15192

### Manual testing scenarios
Module Manager doesn't show module grid when going through below step:
* System > Tools > Web Setup Wizard > Module Manager

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: setup/src/Magento/Setup/Model/Grid/Module.php [hunk 1]
```diff
@@ -178,7 +178,7 @@ private function getModulesInfo(array $moduleList)
     private function addGeneralInfo(array $items)
     {
         foreach ($items as &$item) {
-            $item['moduleName'] = $item['moduleName'] ?: $this->packageInfo->getModuleName($item['name']);
+            $item['moduleName'] = isset($item['moduleName']) ? $item['moduleName'] : $this->packageInfo->getModuleName($item['name']);
```
Comment (rogyar) [inline setup/src/Magento/Setup/Model/Grid/Module.php]: The line length exceeds 120 characters. According to PSR-2 coding style, a line length must be  <= 120 characters. Also as an improvement, I would recommend using null coalescing operator to simplify the construction (http://php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce)

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by rogyar** [comment]
Hello @flancer64. Thank you for your collaboration. Please, consider fixing issues mentioned in the review. Also, adjust the PR description using the template you have upon the PR creation which includes Fixed Issues, Steps to Reproduce and other required sections. 
Thank you

**Comment by flancer64** [comment]
Hi, guys. Thanks for your responses. Feel free to change this code as you wish. It is just my proposal to fix the bug.

**Comment by magento-engcom-team** [comment]
@flancer64 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by hostep** [comment]
@flancer64: can you edit your opening post, follow the [template](https://github.com/magento/magento2/blob/2.2-develop/.github/PULL_REQUEST_TEMPLATE.md) and add issue https://github.com/magento/magento2/issues/15192 under the `Fixed Issues (if relevant)` so that issue automatically gets marked as fixed when this PR gets approved.

Thanks for the fix!

**Comment by flancer64** [comment]
@hostep , done.

**Comment by flancer64** [comment]
@magento-engcom-team , done.

**Comment by magento-engcom-team** [comment]
Hi @flancer64. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.5 release.

Please, consider porting this solution across release lines.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Comment by Donnerberg** [comment]
I am having same problem Screenshoot: http://prntscr.com/jojvx0

But changing module.php line 181 did not solved my problem.

Does any1 have any other idea ?

Many thanks.
Nikola

**Comment by flancer64** [comment]
PHP version? `??` is a 7.x feature. Try to get error message from web server error log, not from browser console.

**Comment by Donnerberg** [comment]
There is nothing in log files: http://prntscr.com/jokvek
( At least nothing when I press Module Manager button )

My PHP version is 7.0. I can upgrade it to version 7.1 if I need.

**Comment by flancer64** [comment]
No, not in Magento logs. Try to look up for error message in `/var/log/[apache2|nginx]/your_host_error.log`

**Comment by Donnerberg** [comment]
Hm,

I'm going to find apache log files.

For now I have found only this: http://prntscr.com/jolgx0

**Comment by flancer64** [comment]
Does the file `/usr/home/donner/www_logs/donnerber.net/error.log` exist?

**Comment by Donnerberg** [comment]
[error.log.20180530.gz](https://github.com/magento/magento2/files/2057299/error.log.20180530.gz)
I found log files at exactly location.

Before I downloaded log files, I have initiated Module Manager a few times.

Is there anything which could tell where is the problem ?

**Comment by flancer64** [comment]
Sorry, @Donnerberg , I see nothing in error log. Could you give me an access log? We need to found request with HTTP result code 500.

**Comment by Donnerberg** [comment]
[access.log.20180530.gz](https://github.com/magento/magento2/files/2057907/access.log.20180530.gz)

Oh, I should send you both. I hope there is something what could help.

Many thanks for helping me with this.

1day older access log file you could find here:
[access.log.20180529.gz](https://github.com/magento/magento2/files/2057918/access.log.20180529.gz)

**Comment by flancer64** [comment]
Sorry, @Donnerberg . I see that these error log & access log are from one server but I cannot see the error message in the log. Did you try to go directly to the URL "http://donnerberg.net.dedivirt595.your-server.de/setup/index.php/module-grid" under admin account (authorized)? You should see smth. like

image

If you will not see & will get error 500 try to debug app. Set breakpoint in `./setup/src/Magento/Setup/Controller/ModuleGrid.php` (if you will go down to this class without errors).

**Comment by Donnerberg** [comment]
Yes you were right: http://prntscr.com/jp0tpm

But I am not sure how to set breakpoint in ./setup/src/Magento/Setup/Controller/ModuleGrid.php  ?

**Comment by flancer64** [comment]
I suppose, you can't if you ask me about it. Perhaps, nobody can, if this server is not under your full control. The last thing I can advise to you - try to look into the HTTP response (this is Dev tools in Chrome - Ctrl + Shift + I):
image

**Comment by Donnerberg** [comment]
Dear Flancer,

I have full access SSH and FTP control of server.

This is error from Network:  http://prntscr.com/jp1vk8

What do you think about that ?

**Comment by flancer64** [comment]
@Donnerberg , do you have "root" access?

You need to click on entry with error (red one) then go to tab "Response".

**Comment by Donnerberg** [comment]
@flancer64 , Yes I have full root access.

Here is screenshoot of response tab: http://prntscr.com/jp2f8u

**Comment by flancer64** [comment]
Try to setup [remote debugging](https://xdebug.org/docs/remote) then walk through the code starting from `./setup/index.php`

**Pull request merged**